### PR TITLE
Refactor tests out of packaged module for dns plugins

### PR DESCRIPTION
**Pull request by ohemorange**
Part of #7593.

```
modify_item () {
    mkdir certbot-dns-$1/tests
    mv certbot-dns-$1/certbot_dns_$1/dns_$1_test.py certbot-dns-$1/tests
    echo "recursive-include tests *" | cat certbot-dns-$1/MANIFEST.in - > /tmp/tempfileone
    mv /tmp/tempfileone certbot-dns-$1/MANIFEST.in
    git add -A
    git commit -m "Refactor tests out of module for certbot-dns-$1"
}
```

Release seems to be working, as is full test suite, and `scripts/test_leauto_upgrades.sh`.

To test `python setup.py test`:
```

for item in cloudflare cloudxns digitalocean dnsimple dnsmadeeasy gehirn google linode luadns nsone ovh rfc2136 sakuracloud route53
do
    cd certbot-dns-$item
    python setup.py test
    cd ~-
done
```

### CODE REVIEW

File: certbot-dns-cloudflare/MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1 @@
-include LICENSE.txt
```
[the remark this answers is not in the document]
Comment (ohemorange) [reply certbot-dns-cloudflare/MANIFEST.in]: whatttttt I swear when I tested it, it appended it.... hm....

[the remark this answers is not in the document]
Comment (ohemorange) [reply certbot-dns-cloudflare/MANIFEST.in]: figured it out.

**Comment by ohemorange** [comment]
Thanks for figuring that out and testing it!

**Comment by ohemorange** [comment]
We probably want that for `certbot` as well.

**Comment by bmw** [comment]
Adding it to Certbot is probably a good idea too but I personally think we're less likely to have a problem there because the includes are a little different:
```
recursive-include certbot/tests/testdata *
recursive-include tests *.py
```

**Comment by ohemorange** [comment]
> Adding it to Certbot is probably a good idea too but I personally think we're less likely to have a problem there because the includes are a little different:

Oh, I guess I could also have done it that way for these. Seems like either way is working, and I don't actually have a preference between them.

**Comment by bmw** [comment]
I don't think it matters much but I personally prefer things as is (maybe with the addition of these excludes to Certbot for consistency and to avoid ever somehow including those files). I think Certbot's a bit of a special case because the `testdata` isn't inside the `tests` directory.

**Pull request merged**